src/ActualLab.Rpc/RpcClientPeerReconnectDelayer.cs

```diff
@@ -23,6 +23,16 @@ public virtual RetryDelay GetDelay(
         RpcClientPeer peer, int tryIndex, Exception? lastError,
         CancellationToken cancellationToken = default)
     {
+        var limits = Hub.Limits;
```

Comment:
Same issue, IMO GetConnection should be responsible for peer termination on "terminal" disconnect, at least that's how I designed it.